src/scanpy/metrics/_common.py

```diff
@@ -19,19 +19,19 @@
     from numpy.typing import NDArray
 
     T_NonSparse = TypeVar("T_NonSparse", bound=NDArray | DaskArray)
-    V = TypeVar("V", bound=NDArray | sparse.csr_matrix)
+    V = TypeVar("V", bound=NDArray | CSRBase)
 
-    _Vals = NDArray | sparse.spmatrix | DaskArray | pd.DataFrame | pd.Series
+    _Vals = NDArray | SpBase | DaskArray | pd.DataFrame | pd.Series
```

Comment:
this and `_resolve_vals` are only places SpBase appears in this file, because we convert to CSR in `_resolve_vals`.